index.js

```diff
@@ -3582,6 +3582,43 @@
     impl: zipWith
   };
 
+  //# intercalate :: (Foldable f, Monoid m) => TypeRep m -> m -> f m -> m
+  //.
+  //. Insert a monoid value between every two values in a list
+  //. containing the same monoid, then flatten the list into
+  //. a single value of the same monoid.
```

Comment:
The description above has a few shortcomings:

  - it does not reference `joinWith`;
  - it references lists; and
  - it describes a two-step algorithm, but we are free to fuse these steps (and we do).

I propose the following:

```javascript
  //. Concatenate the elements of the given structure, separating each pair
  //. of adjacent elements with the given separator.
  //.
  //. See also [`joinWith`](#joinWith).
```